Sources/Test/SimpleTesting/Common.cs

```diff
@@ -0,0 +1,388 @@
+        {
+            Invariant.IsNotNull(input, "input");
+            return input.OrderBy(v => v.SyncTime).ToArray().ToStreamable();
+        }
+
+        public static IStreamable<Empty, TPayload> ToStreamable<TPayload>(this StreamEvent<TPayload>[] input)
+        {
+            Invariant.IsNotNull(input, "input");
+
+            return input.ToObservable()
+                .ToStreamable(null, FlushPolicy.FlushOnPunctuation);
+        }
+
+        public static IEnumerable<T> TrillWhere<T>(IEnumerable<T> enumerable, Expression<Func<T, bool>> wherePredicate, bool disableFusion = true)
+        {
+            var temp = enumerable.ToStreamable();
+            if (disableFusion) temp = temp.AlterEventLifetime(s => s, StreamEvent.InfinitySyncTime);
+            var result = temp.Where(wherePredicate)
+                .ToPayloadEnumerable();
+            return result;
+        }
+        public static bool TestWhere<T>(this IEnumerable<T> enumerable, Expression<Func<T, bool>> wherePredicate, bool disableFusion = true)
+        {
+            var linqResult = enumerable.Where(wherePredicate.Compile());
+            var streamResult = TrillWhere(enumerable, wherePredicate, disableFusion);
+            return linqResult.SequenceEqual(streamResult);
+        }
+
+        public static IEnumerable<U> TrillSelect<T, U>(IEnumerable<T> enumerable, Expression<Func<T, U>> selectFunction, bool disableFusion = true)
+        {
+            var temp = enumerable.ToStreamable();
+            if (disableFusion) temp = temp.AlterEventLifetime(s => s, StreamEvent.InfinitySyncTime);
+            var result = temp.Select(selectFunction)
+                .ToPayloadEnumerable();
+            return result;
+        }
+
+        public static bool TestSelect<T, U>(this IEnumerable<T> enumerable, Expression<Func<T, U>> selectFunction, bool disableFusion = true)
+        {
+            var linqResult = enumerable.Select(selectFunction.Compile());
+            var streamResult = TrillSelect(enumerable, selectFunction, disableFusion);
+            var a = streamResult.ToArray();
+            var b = linqResult.ToArray();
+            return a.SequenceEqual(b);
+        }
+
+        public static bool TestWhereSelect<T, U>(this IEnumerable<T> enumerable, Expression<Func<T, bool>> wherePredicate, Expression<Func<T, U>> selectFunction, bool disableFusion = true)
+        {
+            var linqResult = enumerable.Where(wherePredicate.Compile()).Select(selectFunction.Compile());
+            var trillTemp = enumerable.ToStreamable();
+            if (disableFusion) trillTemp = trillTemp.AlterEventLifetime(s => s, StreamEvent.InfinitySyncTime);
+            var streamResult = trillTemp.Where(wherePredicate)
+                .Select(selectFunction)
+                .ToPayloadEnumerable();
+            return linqResult.SequenceEqual(streamResult);
+        }
+
+        public static IEnumerable<long> CountToInfinity(this long i) { var x = i; while (true) yield return x++; }
```

Comment:
nit: add empty line after function, and probably some inside
(`Enumerable.Range(0, int.MaxValue)` also can be used if items are less than int.MaxValue)

Reply:
Sure. FYI these are just copied/pasted.

---
In reply to: [386614803](https://github.com/microsoft/Trill/pull/131#discussion_r386614803) [](ancestors = 386614803)